registrations/models.py

```diff
@@ -946,6 +946,40 @@ def create_web_store_payment(self):
 
         return SignUpPayment.objects.create(**kwargs)
 
+    def create_web_store_refund(self, payment):
+        orders_data = [
+            {
+                "orderId": payment.external_order_id,
+                "items": [
+                    price_group.to_web_store_partial_refund_json()
+                    for price_group in self.price_groups
+                ],
+            }
+        ]
+        resp_json = create_web_store_refunds(orders_data)
+
+        if isinstance(self, SignUp):
+            signup_relation_kwarg = {"signup": self}
+        else:
+            signup_relation_kwarg = {"signup_group": self}
+        SignUpPaymentRefund.objects.create(
+            **signup_relation_kwarg,
+            payment=payment,
+            amount=self.total_payment_amount,
+            external_refund_id=resp_json["refunds"][0]["refundId"],
```

Comment:
there's a potential KeyError or `IndexError` if the response data is wonky somehow. 
What I understood this is part of call loop in deletion and it does not catch `KeyError` nor `IndexError`

Reply:
Added logic to catch this, log it with `logger` and raise a `WebStoreRefundValidationError` to aid quite a bit with debugging.